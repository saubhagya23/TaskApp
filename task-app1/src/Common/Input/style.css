.input-box {
  height: 30px;
  width: 240px;
  border-radius: 10px;
}