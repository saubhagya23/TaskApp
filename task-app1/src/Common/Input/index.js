import React, { Component } from 'react';
import './style.css';

class Input extends Component {
  render() {
    const {
      type,
      name,
      value,
      placeholder,
      inputStyle,
      onChange,
    } = this.props;
    return(
      <input
        type={type}
        name={name}
        value={value}
        placeholder={placeholder}
        className={`input-box ${inputStyle && inputStyle}`}
        onChange={onChange}
        data-testid="input"
      />
    );
  }
}

export default Input;