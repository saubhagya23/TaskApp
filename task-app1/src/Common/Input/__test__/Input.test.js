import React from 'react';
import ReactDOM from 'react-dom';
import Input from './../index';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

afterEach(cleanup);
it("renders input without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Input/>, div)
})

it('renders input correctly', () => {
  const {getByTestId} = render(<Input type="email"/>);
  expect(getByTestId('input')).toHaveValue("");
})

it("matches snapshot", () => {
  const inputTree = renderer.create(<Input btnText="click me"></Input>).toJSON();
  expect(inputTree).toMatchSnapshot();
})