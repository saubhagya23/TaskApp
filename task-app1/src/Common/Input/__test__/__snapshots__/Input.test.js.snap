exports[`test matches snapshot 1`] = `
<input
  className="input-box undefined"
  data-testid="input"
  name={undefined}
  onChange={undefined}
  placeholder={undefined}
  type={undefined}
  value={undefined} />
`;
